hyperspy/_components/physical_background.py

```diff
@@ -0,0 +1,205 @@
+    b=(model._signal.axes_manager.signal_axes[-1].axis)-0.035
+    Ac=np.interp(b,t,Ac)
+    
+    return Ac
+
+
+def Windowabsorption(model,detector): # this function interpolate the detector efficiency based on dictionnary (create from personnal data) and the signal length. This correspond to the Window parameter  
+ 
+    a=np.array(detector_efficiency[detector])
+    b=(model._signal.axes_manager.signal_axes[-1].axis)-0.035
+    x =a[:,0]
+    y = a[:,1]
+    Accc=np.interp(b, x, y)
+    return Accc
+        
+class Physical_background(Component):
+
+    """
+    Background component based on kramer's law and absorption coefficients
+    Attributes
+    ----------
+    a : float
+    b : float
+    E0 : int 
+    """
+
+    def __init__(self, E0, detector, quantification, coefficients=1, Window=0, quanti=0):
+        Component.__init__(self,['coefficients','E0','Window','quanti'])
+
+        self.coefficients._number_of_elements = 2
+        self.coefficients.value = np.ones(2)
+        
+        self.E0.value=E0
+        
+        self._whitelist['quanti'] = quantification
+        self._whitelist['detector'] = detector       
+        self.quanti.value=quanti
+
+        self.Window.value=Window
+
+        self.E0.free=False
+        self.coefficients.free=True
+        self.Window.free=False
+        self.quanti.free=False
+        
+        self.isbackground=True
+
+        # Boundaries
+        self.coefficients.bmin=0
+        self.coefficients.bmax=None
+
+    def initialyze(self): # this function is necessary to initialyze the quant map
+
+        E0=self.E0.value
+
+        self.quanti._number_of_elements=len(self.model._signal.metadata.Sample.elements)
+        self.quanti._create_array()
+
+        if len(self.model.axes_manager.shape)==1:
+            self.quanti.value=Wpercent(self.model,E0,self._whitelist['quanti'])
+        else: 
+            self.quanti.map['values'][:] = Wpercent(self.model,E0,self._whitelist['quanti'])
+            self.quanti.map['is_set'][:] = True
+        
+        self.Window._number_of_elements=len(self.model._signal.axes_manager.signal_axes[-1].axis)
+        self.Window._create_array()
+        self.Window.value=Windowabsorption(self.model,self._whitelist['detector'])
+        
+        return {'Quant map has been created'}
+        
+    def function(self,x):
+ 
+        b=self.coefficients.value[0]
+        a=self.coefficients.value[1]
+        
+        E0=self.E0.value
+
+        Mu=Mucoef(self.model,self.quanti.value)
+        
+        Mu=np.array(Mu,dtype=float)
+        Mu=Mu[self.model.channel_switches]
+        
+        Window=np.array(self.Window.value,dtype=float)
+        Window=Window[self.model.channel_switches]
+            
+        return np.where((x>0.17) & (x<(E0)),((a*100*((E0-x)/x))*((1-np.exp(-2*Mu*b*10**-5 ))/((2*Mu*b*10**-5)))*Window),0) #implement choice for coating correction
```

Comment:
I'd recommend tidying this return statement (separate it into variables) so that the code is more readable.